iosCalorieTrackrSprintChallenge/iosCalorieTrackrSprintChallenge/ViewControllers/CalorieTrackRViewController.swift

```diff
@@ -0,0 +1,118 @@
+    
+    var calorieIntakeArray: [CalorieIntake] {
+        let fetchRequest: NSFetchRequest<CalorieIntake> = CalorieIntake.fetchRequest()
+        let context = CoreDataStack.shared.mainContext
+        do {
+            return try context.fetch(fetchRequest)
+        } catch {
+            print("error")
+            return []
+        }
+    }
+    
+    // MARK: - Chart Implementation
+    var caloriesAdded: CalorieIntake?
+    var calories: [CalorieIntake] = []
+    var chartNumbers: [Double] = []
+    
+    func setupChart() {
+        chartView.axesColor = .systemTeal
+        chartView.gridColor = .systemRed
+        chartView.backgroundColor = .systemBackground
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.reloadData()
+        setupChart()
+        createObserver()
+        
+    }
+    
+    // MARK: - Observer
+    func createObserver() {
+        NotificationCenter.default.addObserver(self, selector: #selector(refreshViews(notification:)), name: .doneWasTapped, object: nil)
+        print("OBSERVING")
+    }
+   
+    @objc func refreshViews(notification: Notification) {
+        tableView.reloadData()
+    }
+    
+    // MARK: - Functions
+    
+    @IBAction func addTapped(_ sender: Any) {
+        showTextViewAlert()
+    }
+    
+    @objc func showTextViewAlert() {
+        let alertView = UIAlertController(title: "Add Calorie Intake", message: "Enter The Amount Of Calories In The Field", preferredStyle: .alert)
+        
+        alertView.addTextField(configurationHandler: nil)
+        alertView.textFields![0].placeholder = "Calories"
+        alertView.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+        alertView.addAction(UIAlertAction(title: "Done", style: .default, handler: { _ in
+            
+            NotificationCenter.default.post(name: .doneWasTapped, object: nil)
+            
+            guard let calorieInput = Double(alertView.textFields![0].text!), !calorieInput.isZero else { return }
+            let date = Date()
+            print("DONE WAS TAPPED! ENTERED \(calorieInput) CALORIES AT \(Date())")
+            
+            let calorie = CalorieIntake(calories: Double(calorieInput), timestamp: date)
+            self.tableView.reloadData()
+            let calorieNumber = calorie.calories
+            self.chartNumbers.append(calorieNumber)
+            let series = ChartSeries(self.chartNumbers)
+            self.chartView.add(series)
+            
+            do {
+                try CoreDataStack.shared.mainContext.save()
+            } catch {
+                print("error try to save")
+            }
+        }
+        ))
+        let textfield = alertView.textFields![0] as UITextField
+        textfield.keyboardType = UIKeyboardType.numberPad
+        
+        self.present(alertView, animated: true, completion: nil)
+    }
+}
```

Comment:
So it might make sense to store the text field in a property since you seem to be accessing it multiple times.
You should also be using your array of CalorieIntake at the top to get the calories and add them to the chart each time you add a new calorie entry. but you can do this in the selector which gets called from observing the notification.